comm/src/main/scala/coop/rchain/comm/transport/GrpcTransportReceiver.scala

```diff
@@ -43,31 +50,46 @@ object GrpcTransportReceiver {
             .fold(internalServerError("protocol not available in request").pure[Task]) { protocol =>
               rPConfAsk.reader(_.local) >>= (
                   src =>
-                    Task.delay(tellBuffer.pushNext(Send(protocol))).map {
-                      case false => internalServerError("message dropped")
-                      case true  => noResponse(src)
-                    }
+                    Task
+                      .delay(tellBuffer.pushNext(Send(protocol)))
+                      .ifM(
+                        metrics.incrementCounter("enqueued.messages") >> Task
+                          .delay(noResponse(src)),
+                        metrics.incrementCounter("dropped.messages") >> Task
+                          .delay(internalServerError("message dropped"))
+                      )
                 )
             }
 
+        private val circuitBreaker: StreamHandler.CircuitBreaker = streamed =>
+          if (streamed.header.exists(_.networkId != networkId))
+            Opened(StreamHandler.StreamError.wrongNetworkId)
+          else if (streamed.readSoFar > maxStreamMessageSize)
+            Opened(StreamHandler.StreamError.circuitOpened)
+          else Closed
+
         def stream(observable: Observable[Chunk]): Task[ChunkResponse] = {
           import StreamHandler._
           import StreamError.StreamErrorToMessage
 
-          val circuitBreaker: StreamHandler.CircuitBreaker = streamed =>
-            if (streamed.header.map(_.networkId != networkId).getOrElse(false))
-              Opened(StreamHandler.StreamError.wrongNetworkId)
-            else if (streamed.readSoFar > maxStreamMessageSize)
-              Opened(StreamHandler.StreamError.circuitOpened)
-            else Closed
-
           (handleStream(tempFolder, observable, circuitBreaker) >>= {
             case Left(error @ StreamError.Unexpected(t)) => logger.error(error.message, t)
             case Left(error)                             => logger.warn(error.message)
             case Right(msg) =>
-              Task.delay(blobBuffer.pushNext(msg)) >> logger.debug(
-                s"Enqueued for handling packet ${msg.path}"
-              )
+              metrics.incrementCounter("received.packets") >>
+                Task
+                  .delay(blobBuffer.pushNext(msg))
+                  .ifM(
+                    List(
+                      logger.debug(s"Enqueued for handling packet ${msg.path}"),
+                      metrics.incrementCounter("enqueued.packets")
+                    ).sequence,
+                    List(
+                      logger.debug(s"Dropped packet ${msg.path}"),
```

Comment:
Could you attach information from `msg` (such as `sender`) to these messages? It'd be best if network layer had its own logging mechanism so that you wouldn't have to put it into arguments on each log call. The attached peer info could possibly be structured via https://logback.qos.ch/manual/mdc.html.